packages/web-extension/src/content-scripts/player/components/SkipButton.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <RaisedButton v-if="showSkipButton" dark @click.stop="skip"
+    >Skip {{ beforeNextTimeStampTitle }}</RaisedButton
+  >
+</template>
+
+<script lang="ts" setup>
+import { useTimeout } from '@anime-skip/ui';
+import { useGeneralPreferences } from '~/common/state/useGeneralPreferences';
+import { PLAYER_ACTIVITY_TIMEOUT, TIMESTAMP_TYPES } from '~/common/utils/constants';
+import Utils from '~/common/utils/GeneralUtils';
+import { useDisplayedTimestamps } from '../hooks/useDisplayedTimestamps';
+import { useDuration, useVideoController, useVideoState } from '../state/useVideoState';
+
+const { setCurrentTime, setActive, setInactive } = useVideoController();
+const videoState = useVideoState();
+const isActive = computed(() => videoState.isActive);
+const isPaused = computed(() => videoState.isPaused);
+const currentTime = computed(() => videoState.currentTime);
+const preferences = useGeneralPreferences();
+const activeTimestamps = useDisplayedTimestamps();
+const duration = useDuration(videoState);
+const isVideoOver = computed(() => Math.abs(duration.value - currentTime.value) < 0.1);
+
+// Timestamps
+
+const currentTimestamp = computed(() =>
+  Utils.previousTimestamp(currentTime.value, activeTimestamps.value, undefined, -0.1)
+);
+const nextTimestamp = computed(() =>
+  Utils.nextTimestamp(currentTime.value, activeTimestamps.value, preferences.value, true)
+);
+const beforeNextTimeStampTitle = computed(() => {
+  if (!nextTimestamp.value) return;
+  const beforeNextTimeStamp = Utils.previousTimestamp(
+    nextTimestamp.value?.at,
+    activeTimestamps.value,
+    undefined
+  );
+
+  return TIMESTAMP_TYPES.find(ttype => ttype.id === beforeNextTimeStamp?.typeId)?.name;
+});
```

Comment:
Rather than doing this lookup for the first timestamp of the area we're skipping, lets just show the current timestamp's title.

```suggestion
const currentTimestampTitle = computed(() => {
  if (!currentTimestamp.value) return;
  return TIMESTAMP_TYPES[currentTimestamp.value.typeId].name;
});
```

Reply:
I am happy to do so if you are but just in case i'll say that the reason I did it this way because of what we said here: https://github.com/anime-skip/player/issues/172#issuecomment-1053502748, namely that if we are skipping to adjacent sections, the button should say the name of the last section we are skipping (why I chose the name beforeNextTimeStampTitle)

Reply:
Ah my bad, I though that intro was the first timestamp from your comment, not the last. In that case, it should say "Skip Recap" when the recap is going, and "Skip Intro" when the intro is playing

Reply:
I think I made this change already too